docs/source/configuration/slots.md

```diff
@@ -0,0 +1,407 @@
+
+To illustrate how slots are structured and work, let's register a slot component, where the component is `PageHeader`, and the predicate matches a route that begins with `/de/about`.
+
+```ts
+config.registerSlotComponent({
+  slot: 'aboveContent',
+  name: 'header',
+  component: PageHeader,
+  predicates: [RouteCondition('/de/about')],
+});
+```
+
+The following tree structure diagram illustrates the resultant registration.
+
+```text
+Slot (`name`=`aboveContent`)
+└── SlotComponent
+    ├── `slot`=`aboveContent`
+    ├── `name`=`header`
+    ├── `component`=PageHeader
+    └── predicate of "only appear under `/de/about`"
+```
+
+Next, let's register another slot component in the same slot, with the same name and component, but with a different predicate where the content type matches either `Document` or `News Item`.
+
+```ts
+config.registerSlotComponent({
+  slot: 'aboveContent',
+  name: 'header',
+  component: PageHeader,
+  predicates: [ContentTypeCondition(['Document', 'News Item'])],
+});
+```
+
+The following tree structure diagram illustrates the result of the second registration.
+
+```text
+Slot (`name`=`aboveContent`)
+├── SlotComponent
+│   ├── `slot`=`aboveContent`
+│   ├── `name`=`header`
+│   ├── `component`=PageHeader
+│   └── predicate of "only appear under `/de/about`"
+└── SlotComponent
+    ├── `slot`=`aboveContent`
+    ├── `name`=`header`
+    ├── `component`=PageHeader
+    └── predicate of "only appear when the content type is either a Document or News Item"
+```
+
+Finally, let's register another slot component in the same slot, with the same name, but with a different component and without a predicate.
+
+```ts
+config.registerSlotComponent({
+  slot: 'aboveContent',
+  name: 'header',
+  component: 'DefaultHeader',
+});
+```
+
+The following tree structure diagram illustrates the result of the third registration.
+
+```text
+Slot (`name`=`aboveContent`)
+├── SlotComponent
+│   ├── `slot`=`aboveContent`
+│   ├── `name`=`header`
+│   ├── `component`=PageHeader
+│   └── predicate of "only appear under `/de/about`"
+├── SlotComponent
+│   ├── `slot`=`aboveContent`
+│   ├── `name`=`header`
+│   ├── `component`=PageHeader
+│   └── predicate of "only appear when the content type is either a Document or News Item"
+└── SlotComponent
+    ├── `slot`=`aboveContent`
+    ├── `name`=`header`
+    └── `component`=`DefaultHeader`
+```
+
+When the slot components with the same name and component under a given slot have all of their predicates return `true`, then that component will render in the slot.
+Else, if there are slot components with the same name, but with a different component, and without predicates, then that different component will render.
+
+Thus the example slot renderer will have the following behavior.
+
+-   When both a user visits the route beginning with `/de/about`, and the content type is either a Document or News Item, then the component `PageHeader` will render in the `aboveContent` slot.
```

Comment:
No. PageHeader will be used if the route starts with `/de/about` OR if the content type is Document or News Item. Matching either condition is sufficient; they don't have to both be true.